handlers/Subscribe.cpp

```diff
@@ -0,0 +1,233 @@
+
+        if (s == "ledger")
+            manager.subLedger(session);
+        else if (s == "transactions")
+            manager.subTransactions(session);
+        else
+            assert(false);
+    }
+}
+
+void
+unsubscribeToStreams(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& streams = request.at("streams").as_array();
+
+    for (auto const& stream : streams)
+    {
+        std::string s = stream.as_string().c_str();
+
+        if (s == "ledger")
+            manager.unsubLedger(session);
+        else if (s == "transactions")
+            manager.unsubTransactions(session);
+        else
+            assert(false);
+    }
+}
+
+boost::json::value
+validateAccounts(boost::json::object const& request)
+{
+    if (!request.at("accounts").is_array())
+    {
+        return "accounts must be array";
+    }
+
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        if (!account.is_string())
+        {
+            return "account must be strings";
+        }
+
+        std::string s = account.as_string().c_str();
+        auto id = ripple::parseBase58<ripple::AccountID>(s);
+
+        if (!id)
+        {
+            return boost::json::string("invalid account " + s);
+        }
+    }
+
+    return nullptr;
+}
+
+void
+subscribeToAccounts(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        std::string s = account.as_string().c_str();
+
+        auto accountID = ripple::parseBase58<ripple::AccountID>(s);
+
+        if(!accountID)
+        {
+            assert(false);
+            continue;
+        }
+
+        manager.subAccount(*accountID, session);
+    }
+}
+
+void
+unsubscribeToAccounts(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::array const& accounts = request.at("accounts").as_array();
+
+    for (auto const& account : accounts)
+    {
+        std::string s = account.as_string().c_str();
+
+        auto accountID = ripple::parseBase58<ripple::AccountID>(s);
+
+        if(!accountID)
+        {
+            assert(false);
+            continue;
+        }
+
+        manager.unsubAccount(*accountID, session);
+    }
+}
+
+boost::json::object
+doSubscribe(
+    boost::json::object const& request,
+    std::shared_ptr<session>& session,
+    SubscriptionManager& manager)
+{
+    boost::json::object response;
+
+    if (request.contains("streams"))
+    {
+        boost::json::value error = validateStreams(request);
+
+        if(!error.is_null())
+        {
+            response["error"] = error;
+            return response;
+        }   
+    }
```

Comment:
Let's combine this `if` body with the one starting on line 185. Same for the `if` body on line `174` and the one on line `188`

Reply:
See comment below